### PR TITLE
Update "Docs" navigation. Add Spring Security SAML guidance

### PR DESCRIPTION
**Pull request by joelfranusic-okta**
- Added a "Single Sign-On Guidance" section to Docs
- Changed the name of the "Getting Started" section to "Getting Started with the Okta API" to clarify what it's about
- Moved the "SAML Guidance" document to the more appropriate "Single Sign-On Guidance" section
- Added guidance for Spring Security SAML to the new SSO Guidance section

### CODE REVIEW

**Comment by joelfranusic-okta** [comment]
@karlmcguinness-okta - Updated per our conversation earlier today.

File: docs/getting_started/saml_guidance.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: docs_page
-title: SAML
+title: Single Sign-On into Okta
```
Comment (karlmcguinness-okta) [inline docs/getting_started/saml_guidance.md]: Shouldn't it Single Sign-On with Okta or just Single Sign-On for short?  Single Sign-On into Okta implies setting up Okta with external IdP.

**Comment by karlmcguinness-okta** [comment]
Screenshots might be out of date with new admin refresh, didn't check.

**Comment by joelfranusic-okta** [comment]
Screenshots will indeed be out of date with the admin refresh. I'll update them after the refresh.

**Pull request merged**